.github/workflows/kitos-pubsub.yml

```
@@ -0,0 +1,104 @@
+name: Deploy Kitos PubSub
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment to deploy to'
+        required: true
+        type: choice
+        default: 'Staging'
+        options:
+          - 'Development'
+          - 'Staging'
+          - 'Production'
+
+env:
+  DOCKER_REPO: kitos-pubsub
+  DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+  DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+  SERVER_APP_ROOT_FOLDER: "/home/kitosadmintest/app"
+  PROJECT_APP_FOLDER: "PubSub.Application"
+  PROJECT_TEST_FOLDER: "PubSub.Test"
+  PROJECT_DEPLOYMENT_SUB_FOLDER: "Deployment"
+
+jobs:
+  unit-test:
+    runs-on: ubuntu-latest
+    environment: ${{ github.event.inputs.environment }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      
+      - name: Run unit tests
+        working-directory: ${{ env.PROJECT_TEST_FOLDER }}
+        run: dotnet test --logger trx --results-directory TestResults
+      
+      - name: Upload unit test results
+        uses: actions/upload-artifact@v4
+        with:
+          name: kitos-pubsub-unit-test-results
+          path: ${{ env.PROJECT_TEST_FOLDER }}/TestResults
+          retention-days: 7
+    
+  build-and-push-docker-image:
+    runs-on: ubuntu-latest
+    environment: ${{ github.event.inputs.environment }}
+    needs: unit-test
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Login to Docker Hub
+        run: echo "${{ env.DOCKER_PASSWORD }}" | docker login -u "${{ env.DOCKER_USERNAME }}" --password-stdin
+
+      - name: Build and push PubSub API Docker image
+        run: |
+          TAG=${{ github.event.inputs.environment }}  # Tag images based on environment
+          pwd
+          ls
+          sh ${{ env.PROJECT_APP_FOLDER }}/${{ env.PROJECT_DEPLOYMENT_SUB_FOLDER }}/build_and_push_docker_image.sh ${{ env.DOCKER_USERNAME }} ${{ env.DOCKER_REPO }} $TAG
```

```suggestion
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v3
      - name: Build and push
        uses: docker/build-push-action@v6
        with:
          context: ${{ env.PROJECT_APP_FOLDER }}/${{ env.PROJECT_DEPLOYMENT_SUB_FOLDER }}
          push: true
          tags: ${{ secrets.DOCKER_USERNAME }}/${{ github.event.inputs.environment }}:latest
          secrets: |
            GIT_AUTH_TOKEN=${{ secrets.DOCKER_PASSWORD }}
```

Maybe try use dockers own actions?